kubeflow/trainer/options/kubernetes.py

```diff
@@ -0,0 +1,501 @@
+                    raise ValueError("Each volume_mounts entry must be a dictionary")
+                if "name" not in mount:
+                    raise ValueError("Each volume_mounts entry must have a 'name' key")
+                if not mount.get("name"):
+                    raise ValueError("volume_mounts 'name' must be a non-empty string")
+                if "mountPath" not in mount:
+                    raise ValueError("Each volume_mounts entry must have a 'mountPath' key")
+                mount_path = mount.get("mountPath")
+                if not mount_path or not isinstance(mount_path, str):
+                    raise ValueError("volume_mounts 'mountPath' must be a non-empty string")
+                if not mount_path.startswith("/"):
+                    raise ValueError(
+                        f"volume_mounts 'mountPath' must be an absolute path "
+                        f"(start with /): {mount_path}"
+                    )
+
+
+@dataclass
+class PodSpecOverride:
+    """Configuration for overriding pod template specifications.
+
+    Args:
+        service_account_name: Service account to use for the pods.
+        node_selector: Node selector to place pods on specific nodes.
+        affinity: Affinity rules for pod scheduling.
+        tolerations: Tolerations for pod scheduling.
+        volumes: Volumes to add/merge with the pod.
+        init_containers: Init containers to add/merge with the pod.
+        containers: Containers to add/merge with the pod.
+        scheduling_gates: Scheduling gates for the pods.
+        image_pull_secrets: Image pull secrets for the pods.
+    """
+
+    service_account_name: Optional[str] = None
+    node_selector: Optional[dict[str, str]] = None
+    affinity: Optional[dict] = None
+    tolerations: Optional[list[dict]] = None
+    volumes: Optional[list[dict]] = None
+    init_containers: Optional[list[ContainerOverride]] = None
+    containers: Optional[list[ContainerOverride]] = None
+    scheduling_gates: Optional[list[dict]] = None
+    image_pull_secrets: Optional[list[dict]] = None
+
+
+@dataclass
+class PodTemplateOverride:
+    """Configuration for overriding pod templates for specific job types.
+
+    Args:
+        target_jobs: List of job names to apply the overrides to (e.g., ["node", "launcher"]).
+        metadata: Metadata overrides for the pod template (labels, annotations).
+        spec: Spec overrides for the pod template.
+    """
+
+    target_jobs: list[str]
+    metadata: Optional[dict] = None
+    spec: Optional[PodSpecOverride] = None
+
+
+@dataclass
+class Labels:
+    """Add labels to the TrainJob resource metadata (.metadata.labels).
+
+    Supported backends:
+        - Kubernetes
+
+    Args:
+        labels: Dictionary of label key-value pairs to add to TrainJob metadata.
+    """
+
+    labels: dict[str, str]
+
+    def __call__(
+        self,
+        job_spec: dict[str, Any],
+        trainer: Optional[Union["CustomTrainer", "BuiltinTrainer"]],
```

Comment:
Can you use types in this file as well ?
```suggestion
        trainer: Optional[Union[CustomTrainer, BuiltinTrainer]],
```